src/components/interfaces/IData.ts

```diff
@@ -0,0 +1,10 @@
+type SourcesType = [] | string;
```

Comment:
What do you mean here? If it can be a string or array of strings - it should be string[] | string or Array<string> | string. Array by itself needs to be specified of the types inside (foe example, Array<string | number | MyType>). If you are using [] - it means never[]